code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/AuthIntegrationManager.java

```diff
@@ -0,0 +1,48 @@
+package io.cattle.platform.iaas.api.auth;
+
+import io.cattle.platform.iaas.api.auth.integration.interfaces.AuthConfig;
+import io.cattle.platform.iaas.api.auth.integration.interfaces.AuthConfigManager;
+import io.github.ibuildthecloud.gdapi.factory.SchemaFactory;
+import io.github.ibuildthecloud.gdapi.model.ListOptions;
+import io.github.ibuildthecloud.gdapi.request.resource.impl.AbstractNoOpResourceManager;
+
+import java.util.ArrayList;
+import java.util.Map;
+import javax.inject.Inject;
+
+public class AuthIntegrationManager extends AbstractNoOpResourceManager {
+
+    Map<String, AuthConfigManager> authConfigManagers;
+
+    public Map<String, AuthConfigManager> getAuthConfigManagers() {
+        return authConfigManagers;
+    }
+
+    @Inject
+    public void setAuthConfigManagers(Map<String, AuthConfigManager> authConfigManagers) {
+        this.authConfigManagers = authConfigManagers;
+    }
+
+    @Override
+    public Class<?>[] getTypeClasses() {
+        return new Class<?>[]{AuthConfig.class};
+    }
+
+    @Override
+    public String[] getTypes() {
+        return new String[]{"authconfig"};
```

Comment:
Just FYI, I _think_ you dont need to implement both getTypeClasses and getTypes, just getTypeClasses should suffice, but you don't need to change.